src/Common/src/Interop/Windows/Winsock/Interop.WSARecv.cs

```diff
@@ -11,33 +11,66 @@ internal static partial class Interop
     internal static partial class Winsock
     {
         [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
-        internal static extern SocketError WSARecv(
-            [In] SafeCloseSocket socketHandle,
-            [In] ref WSABuffer buffer,
-            [In] int bufferCount,
-            [Out] out int bytesTransferred,
-            [In, Out] ref SocketFlags socketFlags,
-            [In] SafeHandle overlapped,
-            [In] IntPtr completionRoutine);
+        internal static unsafe extern SocketError WSARecv(
+            SafeCloseSocket socketHandle,
+            WSABuffer* buffer,
+            int bufferCount,
+            out int bytesTransferred,
+            ref SocketFlags socketFlags,
+            SafeNativeOverlapped overlapped,
+            IntPtr completionRoutine);
 
         [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
-        internal static extern SocketError WSARecv(
-            [In] SafeCloseSocket socketHandle,
-            [In, Out] WSABuffer[] buffers,
-            [In] int bufferCount,
-            [Out] out int bytesTransferred,
-            [In, Out] ref SocketFlags socketFlags,
-            [In] SafeHandle overlapped,
-            [In] IntPtr completionRoutine);
+        internal static unsafe extern SocketError WSARecv(
+            IntPtr socketHandle,
+            WSABuffer* buffer,
+            int bufferCount,
+            out int bytesTransferred,
+            ref SocketFlags socketFlags,
+            SafeNativeOverlapped overlapped,
+            IntPtr completionRoutine);
 
-        [DllImport(Interop.Libraries.Ws2_32, SetLastError = true, EntryPoint = "WSARecv")]
-        internal static extern SocketError WSARecv_Blocking(
-            [In] IntPtr socketHandle,
-            [In, Out] WSABuffer[] buffers,
-            [In] int bufferCount,
-            [Out] out int bytesTransferred,
-            [In, Out] ref SocketFlags socketFlags,
-            [In] SafeHandle overlapped,
-            [In] IntPtr completionRoutine);
+        internal static unsafe SocketError WSARecv(
+            SafeCloseSocket socketHandle,
+            ref WSABuffer buffer,
+            int bufferCount,
+            out int bytesTransferred,
+            ref SocketFlags socketFlags,
+            SafeNativeOverlapped overlapped,
+            IntPtr completionRoutine)
+        {
+            WSABuffer localBuffer = buffer;
+            return WSARecv(socketHandle, &localBuffer, bufferCount, out bytesTransferred, ref socketFlags, overlapped, completionRoutine);
+        }
+
+        internal static unsafe SocketError WSARecv(
+            SafeCloseSocket socketHandle,
+            WSABuffer[] buffers,
+            int bufferCount,
+            out int bytesTransferred,
+            ref SocketFlags socketFlags,
+            SafeNativeOverlapped overlapped,
+            IntPtr completionRoutine)
+        {
+            fixed (WSABuffer* buffersPtr = &buffers[0])
```

Comment:
Are we guaranteed that buffers will be non-null?  If it's null, this will NullRef.  Maybe an assert to document that the caller needs to guarantee that?  Or alternatively use ```fixed (WSABuffer* buffersPtr = buffers)``` if it could be null.